auras.py

```diff
@@ -191,54 +236,67 @@ def scaled_value(self, value: int, aura_effect: int, index_handlers: list[str])
-				f'+{num_auras}% Physical Damage Reduction',
+				f'+{sum(int((1 + effect/100) * 1) for effect in aura_counter)}% Physical Damage Reduction',
 				'While there are at least five nearby Allies, you and nearby Allies have Onslaught',
 			]
-		elif len(node_stats) == 5 and node_stats[0] == 'Your Offering Skills also affect you':
+		elif node_name == "Necromancer":
 			return [
 				'// necromancer',
-				f'{num_auras * 2}% increased Attack and Cast Speed',
+				f'{sum(int((1 + effect/100) * 2) for effect in aura_counter)}% increased Attack and Cast Speed',
 			]
-		elif len(node_stats) == 2 and node_stats[0] == 'Auras from your Skills grant +1% Physical Damage Reduction to you and Allies':
+		elif node_name == "Unwavering Faith":
 			return [
 				'// unwavering faith',
-				f'+{num_auras}% Physical Damage Reduction',
-				f'{num_auras * 0.2}% of Life Regenerated per second',
+				f'+{sum(int((1 + effect/100) * 2) for effect in aura_counter)}% Physical Damage Reduction',
+				f'{sum(round((1 + effect/100 * 0.2), 1) for effect in aura_counter)}% of Life Regenerated per second',
 			]
-		elif len(node_stats) == 3 and node_stats[0] == '+20% to all Elemental Resistances':
-			return ['// radiant crusade'] + node_stats[1:]
-		elif len(node_stats) == 4 and node_stats[0] == 'Nearby Allies have 20% increased Attack, Cast and Movement Speed':
+		elif node_name == "Radiant Crusade":
+			return ['// radiant crusade'] + ["Deal 10% more Damage"]
+		elif node_name == "Unwavering Crusade":
 			return [
 				'// unwavering crusade',
 				'20% increased Attack, Cast and Movement Speed',
 				'30% increased Area of Effect',
 			]
-		elif len(node_stats) == 3 and node_stats[0].startswith('Auras from your Skills grant 3% increased Attack and Cast\n'):
+		elif node_name == "Commander of Darkness":
 			return [
 				'// commander of darkness',
-				f'{num_auras * 3}% increased Attack and Cast Speed',
+				f'{sum(int((1 + effect/100) * 3) for effect in aura_counter)}% increased Attack and Cast Speed',
 				'30% increased Damage',
 				'+30% to Elemental Resistances',
 			]
-		elif len(node_stats) == 4 and node_stats[0].startswith('For each nearby corpse, you and nearby Allies Regenerate 0.2% of Energy Shield'):
+		elif node_name == "Essence Glutton":
 			return [
 				'// essence glutton',
 				'For each nearby corpse, you and nearby Allies Regenerate 0.2% of Energy Shield per second, up to 2.0% per second',
 				'For each nearby corpse, you and nearby Allies Regenerate 5 Mana per second, up to 50 per second',
 			]
 
+	def parse_gem_quality(self, gem_name, quality, quality_type):
+		gem_info = self.gem_data[gem_name]
+		additional_effects = []
+		aura_effect_increase = 0
+		if quality > 0:
+			gem_quality_stats = gem_info['static']['quality_stats']
+			for stat in gem_quality_stats:
+				if stat["set"] == quality_type.value:
+					stat["value"] = int(stat["value"] * quality/1000)
+					if stat["id"] in self.text: # if the quality grants an additional effect
```

Comment:
out of curiosity, what's an example of this?

Reply:
phantasmal smite gives additional attack/cast speed increase for example

Reply:
ah!